src/IO/AzureBlobStorage/copyAzureBlobStorageFile.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AZURE_BLOB_STORAGE
+
+#include <Storages/StorageAzureBlobCluster.h>
+#include <Storages/StorageAzureBlob.h>
+#include <Interpreters/threadPoolCallbackRunner.h>
+#include <base/types.h>
+#include <functional>
+#include <memory>
+
+
+namespace DB
+{
+class SeekableReadBuffer;
+
+using CreateReadBuffer = std::function<std::unique_ptr<SeekableReadBuffer>()>;
+
+/// Copies a file from AzureBlobStorage to AzureBlobStorage.
+/// The parameters `src_offset` and `src_size` specify a part in the source to copy.
+void copyAzureBlobStorageFile(
+    MultiVersion<Azure::Storage::Blobs::BlobContainerClient> & src_client,
+    MultiVersion<Azure::Storage::Blobs::BlobContainerClient> & dest_client,
+    const String & src_container,
+    const String & src_blob,
+    size_t src_offset,
+    size_t src_size,
+    const String & dest_container,
```

Comment:
How do you use `src_container` and `dest_container` in this function? Only for logging?

Reply:
ya, looks like its only used for logging as we already have the src_client & dest_client. Will update this to log before and remove these parameters